src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -609,9 +609,28 @@ protected override void SelectionChangedHandler(object sender, NotifyCollectionC
             switch (e.Action)
             {
                 case NotifyCollectionChangedAction.Reset:
-                    Model3DDictionary.Values.
-                        Where(v => v is GeometryModel3D).
-                        Cast<GeometryModel3D>().ToList().ForEach(g => g.SetValue(AttachedProperties.ShowSelectedProperty, false));
+
+                    // When deselecting (by clicking on the canvas), all the highlighted HelixWatch3D previews are switched off
+                    // This results in a scenario whereby the list item in the WatchTree/PreviewBubble is still selected, and its
+                    // labels are still displayed in the preview, but the highlighting has been switched off.
+                    // In order to prevent this unintuitive UX behavior, the nodes will first be checked if they are in the 
+                    // nodesSelected dictionary - if they are, they will not be switched off.
+                    var geometryModels = new Dictionary<string, Model3D>();
+                    foreach (var model in Model3DDictionary)
+                    {
+                        var nodePath = model.Key.Contains(':') ? model.Key.Remove(model.Key.IndexOf(':')) : model.Key;
+                        if (model.Value is GeometryModel3D && !nodesSelected.ContainsKey(nodePath))
+                        {
+                            geometryModels.Add(model.Key, model.Value);
+                        }
+                    }
+
+                    var modelValues = geometryModels.Select(x => x.Value);
+
+                    foreach (GeometryModel3D g in modelValues)
```

Comment:
Isn't this a double enumeration over the list? Maybe we can simply enumerate once over `geometryModels` in the `foreach` loop?